syntax = "proto3";

service File {
  // rpc download(FileDownloadReq) returns (stream FileDownloadRsp) {}
  rpc uploadSetName(FileSetNameReq)returns(FileSetNameRsp) {}
  rpc upload(stream FileUploadReq) returns (FileUploadRsp) {}
  // rpc list(ListReq) returns (stream ListRsp) {}
}

message FileDownloadRsp {
  bytes buffer = 1;
}

message FileDownloadReq {
  string name = 1;
}

message FileUploadReq {
  bytes buffer = 1;
}

message FileUploadRsp {
  string result = 1;
}

message ListReq {
}

message FileSetNameReq{
  string name = 1;
  int64 filesize = 2;
}
message FileSetNameRsp{

}

message ListRsp {
  string name = 1;
  int64 size = 2;
}
